usc_mirc.drush.inc

```diff
@@ -329,51 +310,32 @@ EOQ;
     $sandbox['parent_policy_xml'] = $xacml->getXmlString();
     drush_log(dt('Removed POLICY from the collection @pid.', array('@pid' => $parent)));
   }
-  if (!isset($sandbox['total'])) {
-    $sandbox['progress'] = 0;
-    $sandbox['total'] = $tuque->repository->ri->countQuery(format_string($query, array(
-      '!parent' => $parent,
-      '!filter' => '',
-    )), 'sparql');
-    if ($sandbox['total'] == 0) {
-      $context['finished'] = 1;
-      $context['message'] = t('Nothing to lift.');
-      return;
-    }
-    $sandbox['result_stash'] = array();
-  }
   if (empty($sandbox['result_stash'])) {
-    $limited_query = format_string($query, array(
-      '!parent' => $parent,
-      '!filter' => (isset($sandbox['last_date']) ?
-            "FILTER(?created > \"{$sandbox['last_date']}\"^^<xsd:dateTime>)" :
-            ''),
-    )) . <<<EOQ
-LIMIT 1000
-EOQ;
-    drush_log(dt('Querying for 1000 objects.'));
-    $sandbox['result_stash'] = $tuque->repository->ri->sparqlQuery($limited_query);
+    drush_log(dt('Reading file for PIDs.'));
+    module_load_include('inc', 'islandora', 'includes/utilities');
+    $csv = array_map('str_getcsv', file($file));
+    // Munge into our stash.
+    foreach ($csv as $line) {
+      $value = reset($line);
+      $value = str_replace('info:fedora/', '', $value);
+      if (!empty($value) && islandora_is_valid_pid($value)) {
```

Comment:
After coercing into an array of, this could be done with an `array_filter($pids, 'islandora_is_valid_pid');`.

Really:
```php
$to_bare_pid = function ($value) {
  return str_replace('info:fedora/', '', $value);
};
$sandbox['result_stash'] = array_filter(array_map($to_bare_pid, array_map('reset', array_map('str_getcsv', file($file)))), 'islandora_is_valid_pid');
```